src/Provider/AbstractProvider.php

```diff
@@ -135,17 +176,46 @@ public function setScopes(array $scopes)
         $this->scopes = $scopes;
     }
 
+    /**
+     * Get a new random string to use for auth state.
+     *
+     * @param  integer $length
+     * @return string
+     */
+    protected function getRandomState($length = 32)
+    {
+        $generator = $this
+            ->getRandomFactory()
+            ->getMediumStrengthGenerator();
```

Comment:
@ircmaxell we always use the medium generator. By injecting the factory, we allow the user to provide their own source/mixer.